lib/stormpath-sdk/http/http_client_request_executor.rb

```diff
@@ -19,18 +19,17 @@ class HttpClientRequestExecutor
       include Stormpath::Http::Authc
       include Stormpath::Util::Assert
 
-      def initialize(api_key, options = {})
+      def initialize(options = {})
         @signer = Sauthc1Signer.new
-        @api_key = api_key
         @http_client = HTTPClient.new options[:proxy]
       end
 
-      def execute_request(request, redirects_limit = 10)
+      def execute_request(request, api_key, redirects_limit = 10)
```

Comment:
Almost there, but the api_key should be a property of the`Request` object, not a new argument here. I think this change makes sense since all calls to the REST API require an api_key, and also keeps the code cleaner.

Note that the `Sauthc1Signer` class will also need to change to only receive the `Request` object since the api_key will be a property of that class.